addons/binding/org.openhab.binding.smhi/src/main/java/org/openhab/binding/smhi/internal/model/SmhiParameterTables.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.smhi.internal.model;
+
+import java.util.HashMap;
+
+/**
+ * The {@link smhiParameterTables} is the Java class used to map the JSON response to an SMHI
+ * request.
+ *
+ * @author Michael Parment - Initial contribution
+ */
+public class SmhiParameterTables {
```

Comment:
see above. this feels weird. I'll comment on that later.

Reply:
?

Reply:
Have a look [here](https://github.com/openhab/openhab2-addons/blob/4d8a9390c013c16d978ad0cadc8958a34bfb350d/bundles/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/Util.java#L91) and [here](https://github.com/openhab/openhab2-addons/blob/master/bundles/org.openhab.binding.onewire/src/main/resources/accepted_itemtypes.properties).

IMO this is the better way to fill static tables (and easier to extend/understand).

Reply:
When i add your Util.java and add in SmhiBindingConstants.java 
`public static final Map<Integer, String> PCAT_MAP = Util.readPropertiesFile("pcat.properties");`

I receive 
`!ENTRY org.openhab.binding.smhi 4 0 2019-04-25 22:54:00.000
!MESSAGE [org.openhab.binding.smhi.internal.SmhiHandlerFactory(25)] Error during instantiation of the implementation object
!STACK 0
java.lang.NoClassDefFoundError: Could not initialize class org.openhab.binding.smhi.internal.SmhiHandlerFactory
`

Reply:
`Util.readProperties(..)` returns a `Map<String,String>`, you need a `Map<Integer,String>`. You could do

```
public static final Map <Integer, String> PCAT_MAP = Util.readPropertiesFile("pcat.properties").entrySet().stream().collect(Collectors.toMap(e->Integer.parseInt(e.getKey()), Entry::getValue));
```

Reply:
I am doing that.

Reply:
What is unclear here?